cmake/org.eclipse.cdt.cmake.core/src/org/eclipse/cdt/cmake/core/CMakeBuildConfiguration.java

```diff
@@ -124,6 +117,45 @@ public CMakeBuildConfiguration(IBuildConfiguration config, String name, IToolCha
 		this.toolChainFile = toolChainFile;
 	}
 
+	@Override
+	public ICMakeProperties getCMakeProperties() {
+		// Create CMake default properties
+		ICMakeProperties properties = CMakePropertiesFactory.createProperties();
+		// Set defaults for both platforms
+		IOsOverrides windowsOverrides = properties.getWindowsOverrides();
+		windowsOverrides.setDefaultGenerator(CMakeGenerator.Ninja);
+		IOsOverrides linuxOverrides = properties.getLinuxOverrides();
+		linuxOverrides.setDefaultGenerator(CMakeGenerator.Ninja);
+		// reset(true) causes the CMake generator to be reset to it's default value.
+		properties.reset(true);
```

Comment:
What is the purpose of this? ICMakeProperties is a temporary object now as we have removed serialization and any chance of this being used with the UI in its current form.